### PR TITLE
[Keyboard] Add maJIStic keyboard

### PR DESCRIPTION
**Pull request by yossiyossy**
## Description

Added support for the maJIStic, a split JIS keyboard.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Review by drashna — APPROVED** [review]

File: keyboards/maJIStic/rules.mk [hunk 1]
```diff
@@ -0,0 +1,22 @@
+MCU = atmega32u4
+
+BOOTLOADER = atmel-dfu
```
Comment (fauxpark) [inline keyboards/maJIStic/rules.mk:3]:
```suggestion
# MCU name
MCU = atmega32u4

# Bootloader selection
BOOTLOADER = atmel-dfu
```

File: keyboards/maJIStic/rules.mk [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+CONSOLE_ENABLE = yes        # Console for debug
+COMMAND_ENABLE = yes        # Commands for debug and configuration
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = no            # USB Nkey Rollover
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
+BLUETOOTH_ENABLE = no       # Enable Bluetooth
+AUDIO_ENABLE = no           # Audio output on port C6
```
Comment (fauxpark) [inline keyboards/maJIStic/rules.mk:20]:
```suggestion
AUDIO_ENABLE = no           # Audio output
```

File: keyboards/maJIStic/readme.md [hunk 3]
```diff
@@ -0,0 +1,15 @@
+# maJIStic
+
+![maJIStic](https://raw.githubusercontent.com/yossiyossy/maJIStic/master/images/main_image.jpg)
+
+A short description of the keyboard/project
```
Comment (fauxpark) [inline keyboards/maJIStic/readme.md:5]: Please fill this out.

File: keyboards/maJIStic/keymaps/default/keymap.c [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+                     _______, _______, _______, _______, _______, _______,              _______, _______, _______, _______, _______, JP_UNDS, _______, \
+                     _______, _______, _______, _______, _______, _______,              _______, _______, _______, _______, KC_BRMU, KC_VOLU, _______,  \
+                     _______, _______, _______, _______, _______,                       _______,          _______, _______, KC_BRMD, KC_VOLD, _______ \
+    ),
+
+};
+
+bool process_record_user(uint16_t keycode, keyrecord_t *record) {
+    return true;
+}
```
Comment (fauxpark) [inline keyboards/maJIStic/keymaps/default/keymap.c:50]:
```suggestion
    [_BASE] = LAYOUT(
                     JP_1,    JP_2,    JP_3,    JP_4,     JP_5,               JP_6,     JP_7,               JP_8,    JP_9,     JP_0,    JP_MINS, JP_CIRC, JP_YEN,  KC_BSPC,
                     KC_TAB,  KC_Q,    KC_W,    KC_E,     KC_R,               KC_T,     KC_Y,               KC_U,    KC_I,     KC_O,    KC_P,    JP_AT,   JP_LBRC, KC_ENT ,
                     KC_CAPS, KC_A,    KC_S,    KC_D,     KC_F,               KC_G,     KC_H,               KC_J,    KC_K,     KC_L,    JP_SCLN, JP_COLN, JP_RBRC,
                     KC_LSFT, KC_Z,    KC_X,    KC_C,     KC_V,               KC_B,     KC_N,               KC_M,    JP_COMM,  JP_DOT,  JP_SLSH, KC_UP,   KC_RSFT,
                     KC_LCTL, KC_LALT, KC_LGUI, KC_LANG2, LT(_LOWER, KC_SPC),           LT(_LOWER, KC_SPC), KC_LANG1, KC_RGUI, KC_LEFT, KC_DOWN, KC_RIGHT
    ),
    [_LOWER] = LAYOUT(
                     KC_F1,   KC_F2,   KC_F3,   KC_F4,   KC_F5,   KC_F6,                KC_F7,   KC_F8,   KC_F9,   KC_F10,  KC_F11,  KC_F12,  _______, _______,
                     KC_ESC,  _______, _______, _______, _______, _______,              _______, _______, _______, _______, _______, _______, _______, _______,
                     _______, _______, _______, _______, _______, _______,              _______, _______, _______, _______, _______, JP_UNDS, _______,
                     _______, _______, _______, _______, _______, _______,              _______, _______, _______, _______, KC_BRMU, KC_VOLU, _______,
                     _______, _______, _______, _______, _______,                       _______,          _______, _______, KC_BRMD, KC_VOLD, _______
    )
};
```

File: keyboards/maJIStic/config.h [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+#define MATRIX_COLS 8
+
+#define MATRIX_ROW_PINS \
+    { F6, F7, B1, B3, B2 }
+#define MATRIX_COL_PINS \
+    { D4, C6, D7, E6, B4, B5, F4, F5 }  // F4, F5 is dummy
+#define MATRIX_ROW_PINS_RIGHT \
+    { D4, C6, D7, E6, B4 }
+#define MATRIX_COL_PINS_RIGHT \
+    { B6, B2, B3, B1, F7, F6, F5, F4 }
```
Comment (fauxpark) [inline keyboards/maJIStic/config.h:41]:
```suggestion
#define MATRIX_ROW_PINS { F6, F7, B1, B3, B2 }
#define MATRIX_COL_PINS { D4, C6, D7, E6, B4, B5, F4, F5 }  // F4, F5 is dummy
#define MATRIX_ROW_PINS_RIGHT { D4, C6, D7, E6, B4 }
#define MATRIX_COL_PINS_RIGHT { B6, B2, B3, B1, F7, F6, F5, F4 }
```

File: keyboards/maJIStic/config.h [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#pragma once
+
+#include "config_common.h"
+
+/* USB Device descriptor parameter */
+#define VENDOR_ID 0xFEED
+#define PRODUCT_ID 0x0000
```
Comment (fauxpark) [inline keyboards/maJIStic/config.h:25]:
```suggestion
#define PRODUCT_ID 0x6E55
```
```
echo -n "maJIStic" | sha1sum | cut -c1-4
```

**Review by fauxpark — CHANGES_REQUESTED** [review]
Please rename the keyboard folder to lowercase.

File: keyboards/majistic/majistic.h [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+ * layout of the board and position of the keys.
+ *
+ * The second converts the arguments into a two-dimensional array which
+ * represents the switch matrix.
+ */
+
+#define LAYOUT(\
+               l00, l01, l02, l03, l04, l05,     r00, r01, r02, r03, r04, r05, r06, r07, \
+               l10, l11, l12, l13, l14, l15,     r10, r11, r12, r13, r14, r15, r16, r17, \
+               l20, l21, l22, l23, l24, l25,     r20, r21, r22, r23, r24, r25, r26,      \
```
Comment (noroadsleft) [inline keyboards/majistic/majistic.h:33]: QMK custom is that ISO/JIS Enter keys should be on the home row.

```suggestion
               l10, l11, l12, l13, l14, l15,     r10, r11, r12, r13, r14, r15, r16,      \
               l20, l21, l22, l23, l24, l25,     r20, r21, r22, r23, r24, r25, r26, r17, \
```

File: keyboards/majistic/keymaps/default/keymap.c [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+enum custom_keycodes {
+    L_SPC = SAFE_RANGE,
+};
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+    /* Base */
+    [_BASE] = LAYOUT(
+                     JP_1,    JP_2,    JP_3,    JP_4,     JP_5,               JP_6,     JP_7,               JP_8,    JP_9,     JP_0,    JP_MINS, JP_CIRC, JP_YEN,  KC_BSPC, \
+                     KC_TAB,  KC_Q,    KC_W,    KC_E,     KC_R,               KC_T,     KC_Y,               KC_U,    KC_I,     KC_O,    KC_P,    JP_AT,   JP_LBRC, KC_ENT , \
+                     KC_CAPS, KC_A,    KC_S,    KC_D,     KC_F,               KC_G,     KC_H,               KC_J,    KC_K,     KC_L,    JP_SCLN, JP_COLN, JP_RBRC, \
```
Comment (noroadsleft) [inline keyboards/majistic/keymaps/default/keymap.c:34]:
```suggestion
                     KC_TAB,  KC_Q,    KC_W,    KC_E,     KC_R,               KC_T,     KC_Y,               KC_U,    KC_I,     KC_O,    KC_P,    JP_AT,   JP_LBRC,          \
                     KC_CAPS, KC_A,    KC_S,    KC_D,     KC_F,               KC_G,     KC_H,               KC_J,    KC_K,     KC_L,    JP_SCLN, JP_COLN, JP_RBRC, KC_ENT , \
```

File: keyboards/majistic/keymaps/default/keymap.c [hunk 9]
```diff
@@ -0,0 +1,45 @@
...
+                     KC_LSFT, KC_Z,    KC_X,    KC_C,     KC_V,               KC_B,     KC_N,               KC_M,    JP_COMM,  JP_DOT,  JP_SLSH, KC_UP,   KC_RSFT, \
+                     KC_LCTL, KC_LALT, KC_LGUI, KC_LANG2, LT(_LOWER, KC_SPC),           LT(_LOWER, KC_SPC), KC_LANG1, KC_RGUI, KC_LEFT, KC_DOWN, KC_RIGHT \
+    ),
+    [_LOWER] = LAYOUT(
+                     KC_F1,   KC_F2,   KC_F3,   KC_F4,   KC_F5,   KC_F6,                KC_F7,   KC_F8,   KC_F9,   KC_F10,  KC_F11,  KC_F12,  _______, _______, \
+                     KC_ESC,  _______, _______, _______, _______, _______,              _______, _______, _______, _______, _______, _______, _______, _______, \
+                     _______, _______, _______, _______, _______, _______,              _______, _______, _______, _______, _______, JP_UNDS, _______, \
```
Comment (noroadsleft) [inline keyboards/majistic/keymaps/default/keymap.c:41]:
```suggestion
                     KC_ESC,  _______, _______, _______, _______, _______,              _______, _______, _______, _______, _______, _______, _______,          \
                     _______, _______, _______, _______, _______, _______,              _______, _______, _______, _______, _______, JP_UNDS, _______, _______, \
```

File: keyboards/majistic/info.json [hunk 10]
```diff
@@ -0,0 +1,78 @@
...
+								{"label":"R", "x":4, "y":1},
+								{"label":"T", "x":5, "y":1},
+								{"label":"Y", "x":7, "y":1},
+								{"label":"U", "x":8, "y":1},
+								{"label":"I", "x":9, "y":1},
+								{"label":"O", "x":10, "y":1},
+								{"label":"P", "x":11, "y":1},
+								{"label":"@", "x":12, "y":1},
+								{"label":"[", "x":13, "y":1},
+								{"label":"Enter", "x":14.25, "y":1, "w":1.25, "h":2},
```
Comment (noroadsleft) [inline keyboards/majistic/info.json:37]:
```suggestion
```

File: keyboards/majistic/info.json [hunk 11]
```diff
@@ -0,0 +1,78 @@
...
+								{"label":"F", "x":4.25, "y":2},
+								{"label":"G", "x":5.25, "y":2},
+								{"label":"H", "x":7.25, "y":2},
+								{"label":"J", "x":8.25, "y":2},
+								{"label":"K", "x":9.25, "y":2},
+								{"label":"L", "x":10.25, "y":2},
+								{"label":";", "x":11.25, "y":2},
+								{"label":":", "x":12.25, "y":2},
+								{"label":"]", "x":13.25, "y":2},
+								{"label":"Shift", "x":0, "y":3, "w":1.75},
```
Comment (noroadsleft) [inline keyboards/majistic/info.json:51]:
```suggestion
								{"label":"Enter", "x":14.25, "y":1, "w":1.25, "h":2},
								{"label":"Shift", "x":0, "y":3, "w":1.75},
```

**Review by noroadsleft — CHANGES_REQUESTED** [review]

**Comment by yossiyossy** [comment]
@fauxpark @noroadsleft 
Thank you for comments! I fixed all my mistakes, I think.
Could you review again? And if there are any other mistakes, I'm very sorry but, please tell me again!

**Review by noroadsleft — APPROVED** [review]

**Comment by noroadsleft** [comment]
Thanks!

**Pull request merged**